api/lib/block-listeners/cosmos-node-subscription.js

```diff
@@ -204,20 +204,20 @@ class CosmosNodeSubscription {
   }
 
   async addPopularityToValidators(validators, networkId) {
```

Comment:
can be put into store as it doesn't depend on any chain specific data

Reply:
addressed above: perfomance killer for some reason